src/test/scala/com/redislabs/provider/redis/rdd/Person.scala

```diff
@@ -0,0 +1,13 @@
+package com.redislabs.provider.redis.rdd
+
+import org.apache.spark.sql.{Encoder, Encoders}
+
+/**
+  * @author The Viet Nguyen
+  */
+case class Person(name: String, age: Int, address: String, salary: Double)
+
+object Person {
+  implicit val personEncoder: Encoder[Person] = Encoders.product[Person]
```

Comment:
Encoders for case classes are available with ` import spark.implicits._`. We cannot do exactly that because `spark` is a var in the tests. We can overcome it with the trick like this https://github.com/apache/spark/blob/c0949dc944b7e2fc8a4465acc68a8f2713b3fa13/sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestData.scala#L31

Let's create a helper object
```
object testImplicits extends SQLImplicits {
    protected override def _sqlContext: SQLContext = spark.sqlContext
  }
``` 
in Env.scala and then in the test where you need to use datasets, you should be able to import that 
```
import testImplicits._
```
I would also think about importing this implicits only for the tests where you need datasets and don't import it for all tests. In this way we will make sure our dataframe persistence doesn't depend on this implicit

Reply:
thank you, this is interesting and useful